packages/app/src/components/VariationSelector/VariationSelector.tsx

```diff
@@ -0,0 +1,433 @@
+import { useIntersectionObserver } from '@/utils/useIntersectionObserver'
+import { useKeyboardShortcuts } from '@/utils/useKeyboardShortcuts'
+import { Button } from '../Button/Button'
+import { ButtonGroup } from '../Button/ButtonGroup'
+import { DelayedShow } from '../DelayedShow/DelayedShow'
+import { useRequestModal } from '../Modal/ModalContext'
+import { ModalTitleBar } from '../Modal/ModalTitleBar'
+import ui from './VariationSelector.module.css'
+import type { Setter } from 'solid-js'
+import type { v2f } from 'typegpu/data'
+import type {
+  FlameDescriptor,
+  TransformFunction,
+  TransformId,
+  VariationId,
+} from '@/flame/schema/flameSchema'
+import type { TransformVariationDescriptor } from '@/flame/variations'
+import type { ChangeHistory } from '@/utils/createStoreHistory'
+
+const CANCEL = 'cancel'
+
+function PreviewFinalFlame(props: {
+  flame: FlameDescriptor
+  setFlamePosition: Setter<v2f>
+  setFlameZoom: Setter<number>
+}) {
+  return (
+    <Root adapterOptions={{ powerPreference: 'high-performance' }}>
+      <AutoCanvas pixelRatio={1}>
+        <WheelZoomCamera2D
+          zoom={[
+            () => props.flame.renderSettings.camera.zoom,
+            props.setFlameZoom,
+          ]}
+          position={[
+            () => vec2f(...props.flame.renderSettings.camera.position),
+            props.setFlamePosition,
+          ]}
+        >
+          <Flam3
+            quality={0.99}
+            pointCountPerBatch={4e6}
+            adaptiveFilterEnabled={false}
+            flameDescriptor={props.flame}
+            renderInterval={10}
+            onExportImage={undefined}
+            edgeFadeColor={vec4f(0)}
+          />
+        </WheelZoomCamera2D>
+      </AutoCanvas>
+    </Root>
+  )
+}
+
+function Preview(props: { flame: FlameDescriptor }) {
+  return (
+    <Root adapterOptions={{ powerPreference: 'high-performance' }}>
+      <AutoCanvas pixelRatio={1}>
+        <Camera2D
+          position={vec2f(...props.flame.renderSettings.camera.position)}
+          zoom={props.flame.renderSettings.camera.zoom}
+        >
+          <Flam3
+            quality={0.99}
+            pointCountPerBatch={4e6}
+            adaptiveFilterEnabled={false}
+            flameDescriptor={props.flame}
+            renderInterval={10}
+            onExportImage={undefined}
+            edgeFadeColor={vec4f(0)}
+          />
+        </Camera2D>
+      </AutoCanvas>
+    </Root>
+  )
+}
+type respondType = {
+  transform: TransformFunction | typeof CANCEL
+  variation: TransformVariationDescriptor | typeof CANCEL
+}
+type VariationSelectorModalProps = {
+  currentVar: TransformVariationDescriptor
+  currentFlame: FlameDescriptor
+  transformId: TransformId
+  variationId: VariationId
+  respond: (value: respondType) => void
+}
+const lazyLoadAmount = 10
+
+function ShowVariationSelector(props: VariationSelectorModalProps) {
+  const variationPreviewFlames: Record<string, FlameDescriptor> =
```

Comment:
this const can be outside of the component right? Doesn't seem like it needs to be created in each variation selector